src/NServiceBus.NHibernate/Outbox/NHibernateOutbox.cs

```diff
@@ -0,0 +1,158 @@
+namespace NServiceBus.Features
+{
+    using System;
+    using System.Configuration;
+    using System.Transactions;
+    using global::NHibernate.Mapping.ByCode;
+    using global::NHibernate.Transaction;
+    using Microsoft.Extensions.DependencyInjection;
+    using NHibernate.Outbox;
+    using NHibernate.SynchronizedStorage;
+    using NServiceBus.Outbox;
+    using NServiceBus.Outbox.NHibernate;
+    using Persistence.NHibernate;
+    using Configuration = global::NHibernate.Cfg.Configuration;
+    using Environment = global::NHibernate.Cfg.Environment;
+
+    /// <summary>
+    /// NHibernate Outbox feature
+    /// </summary>
+    public class NHibernateOutbox : Feature
+    {
+
+        internal const string OutboxMappingSettingsKey = "NServiceBus.NHibernate.OutboxMapping";
+        internal const string OutboxTableNameSettingsKey = "NServiceBus.NHibernate.OutboxTableName";
+        internal const string OutboxSchemaNameSettingsKey = "NServiceBus.NHibernate.OutboxSchemaName";
+        internal const string OutboxConcurrencyModeSettingsKey = "NServiceBus.NHibernate.OutboxPessimisticMode";
+        internal const string OutboxTransactionModeSettingsKey = "NServiceBus.NHibernate.OutboxTransactionScopeMode";
+        internal const string OutboxTransactionIsolationLevelSettingsKey = "NServiceBus.NHibernate.OutboxTransactionIsolationLevel";
+        internal const string OutboxTransactionScopeModeIsolationLevelSettingsKey = "NServiceBus.NHibernate.OutboxTransactionScopeModeIsolationLevel";
+
+        /// <summary>
+        /// Creates a new instance of the feature
+        /// </summary>
+        public NHibernateOutbox()
+        {
+            Defaults(x => x.EnableFeatureByDefault<NHibernateStorageSession>());
+
+            DependsOn<Outbox>();
+            DependsOn<NHibernateStorageSession>();
+        }
+
+        /// <inheritdoc />
+        protected override void Setup(FeatureConfigurationContext context)
+        {
+            var config = context.Settings.Get<NHibernateConfiguration>();
+
+            var pessimisticMode = context.Settings.GetOrDefault<bool>(OutboxConcurrencyModeSettingsKey);
+            var transactionScopeMode = context.Settings.GetOrDefault<bool>(OutboxTransactionModeSettingsKey);
+            var transactionScopeIsolationLevel = context.Settings.GetOrDefault<IsolationLevel>(OutboxTransactionScopeModeIsolationLevelSettingsKey);
+            var adoIsolationLevel = context.Settings.GetOrDefault<System.Data.IsolationLevel>(OutboxTransactionIsolationLevelSettingsKey);
+            if (adoIsolationLevel == default)
+            {
+                //Default to Read Committed
+                adoIsolationLevel = System.Data.IsolationLevel.ReadCommitted;
```

Comment:
Why the fully qualified type?

Reply:
Because `IsolationLevel` refers to a different namespace